lib/puppet_x/msutter/templates/invoke_dsc_resource.erb

```diff
@@ -42,12 +42,13 @@ $invokeParams = @{
 <% dsc_parameters.each do |p| -%>
     <%= p.name.to_s.gsub(/^dsc_/,'') %> = <%= format_dsc_value(p.value) %>
 <% end -%>
-  }
-<% if resource[:dscmeta_module_name].casecmp('PSDesiredStateConfiguration') == 0 -%>
+  }<% if resource.parameters[:dscmeta_module_version] %>
   ModuleName = @{
-    <%= "ModuleName=\"#{resource[:dscmeta_module_name]}\"" %>
-    <%= "RequiredVersion=\"#{resource[:dscmeta_module_version]}\"" if resource.parameters[:dscmeta_module_version] %>
+    ModuleName      = <%= "\"#{resource[:dscmeta_module_name]}\"" %>
+    RequiredVersion = <%= "\"#{resource[:dscmeta_module_version]}\"" %>
   }
+<% else %>
+  ModuleName = <%= "\"#{resource[:dscmeta_module_name]}\"" %>
 <% end -%>
```

Comment:
Yup, it errors with just a `ModuleName` in there... if you want to specify the parent `ModuleName` as a hash, both `ModuleName` and `RequiredVersion` are required.